x/wasm/keeper/relay.go

```diff
@@ -130,13 +133,34 @@ func (k Keeper) OnRecvPacket(
 	res, gasUsed, execErr := k.wasmVM.IBCPacketReceive(codeInfo.CodeHash, env, msg, prefixStore, cosmwasmAPI, querier, ctx.GasMeter(), gas, costJSONDeserialization)
 	k.consumeRuntimeGas(ctx, gasUsed)
 	if execErr != nil {
-		panic(execErr)
+		panic(execErr) // let contract fully abort IBC receive in certain case
 	}
-	if res.Err != "" { // handle error case as before https://github.com/CosmWasm/wasmvm/commit/c300106fe5c9426a495f8e10821e00a9330c56c6
-		return nil, sdkerrors.Wrap(types.ErrExecuteFailed, res.Err)
+	if res.Err != "" {
+		// return error ACK with non-redacted contract message, state will be reverted
+		return channeltypes.Acknowledgement{
```

Comment:
Kind of surprised to see this type here and a custom one for success.

Why not `type ContractRevertStateAck = string` or something like that?

Reply:
Oh, you use `channeltypes.NewErrorAcknowledgement(err)` in ibc.go above, and this is different?

Reply:
yes, the `NewErrorAcknowledgement` constructor redacts the error message. This impl does not